Source/Charts/Data/Implementations/Standard/ChartDataSet.swift

```diff
@@ -517,3 +490,82 @@ open class ChartDataSet: ChartBaseDataSet
         return copy
     }
 }
+
+// MARK: MutableCollection
+extension ChartDataSet: MutableCollection {
+    public typealias Index = Int
+    public typealias Element = ChartDataEntry
+
+    public var startIndex: Index {
+        return values.startIndex
+    }
+
+    public var endIndex: Index {
+        return values.endIndex
+    }
+
+    public func index(after: Index) -> Index {
+        return values.index(after: after)
+    }
+
+    public subscript(position: Index) -> Element {
+        get {
+            // This is intentionally not a safe subscript to mirror
+            // the behaviour of the built in Swift Collection Types
+            return values[position]
+        }
+        set {
+            isIndirectValuesCall = true
+            calcMinMax(entry: newValue)
+            self.values[position] = newValue
+        }
+    }
+}
+
+// MARK: RandomAccessCollection
+extension ChartDataSet: RandomAccessCollection {
+    public func index(before: Index) -> Index {
+        return values.index(before: before)
+    }
+}
+
+// MARK: RangeReplaceableCollection
+extension ChartDataSet: RangeReplaceableCollection {
+    public func append(_ newElement: Element) {
+        isIndirectValuesCall = true
+        calcMinMax(entry: newElement)
```

Comment:
is it duplicated calls in `addEntry(_ e: ChartDataEntry) -> Bool`? There it already had a `calcMinMax`

Reply:
This doesn't use `addEntry(_:)` so there is no duplicate call

Reply:
No I mean in addEntry(), it will call `append(e)` I assume `append(e)` is this function call?

Reply:
Correct, I will update addEntry to only call this and return `true`